test-data/unit/check-dataclasses.test

```diff
@@ -80,6 +80,29 @@ reveal_type(D)  # N: Revealed type is 'def (a: builtins.int, b: builtins.int, c:
 
 [builtins fixtures/list.pyi]
 
+[case testDataclassesMultipleInheritance]
+from dataclasses import dataclass, field, InitVar
+@dataclass
+class A:
+    a: bool
+
+
+@dataclass
+class B:
+    b: InitVar[bool]
+    _b: bool = field(init=False)
+
+    def __post_init__(self, b: bool):
+        self._b = b
+
+
+@dataclass
+class C(A, B):
+    pass
+
+
```

Comment:
Could you please add `reveal_type(C)` here to be sure the type of constructor is correct?